smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/waiters/WaiterAcceptorGenerator.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.swift.codegen.waiters
+
+import software.amazon.smithy.jmespath.JmespathExpression
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.core.CodegenContext
+import software.amazon.smithy.swift.codegen.utils.toLowerCamelCase
+import software.amazon.smithy.waiters.Acceptor
+import software.amazon.smithy.waiters.Matcher
+import software.amazon.smithy.waiters.PathComparator
+import software.amazon.smithy.waiters.PathMatcher
+
+// Renders the initializer for one Smithy waiter acceptor into Swift,
+// using the provided writer & shapes.
+// The Acceptor is defined in Smithy here:
+// https://smithy.io/2.0/additional-specs/waiters.html#acceptor-structure
+class WaiterAcceptorGenerator(
```

Comment:
Can you add tests for this that verify the generated code is what we expect? That will also be a big help with reviewing this!

Reply:
Done, there are now codegen tests for each type of acceptor, though the expected code will change as JMESPath code generation changes.